src/net.h

```diff
@@ -614,12 +614,27 @@ class CNetMessage {
     int readData(const char *pch, unsigned int nBytes);
 };
 
+/** The TransportSerializer prepares messages for the network transport
+ */
+class TransportSerializer {
+public:
+    // prepare message for transport (insert header, etc.)
+    virtual void prepareForTransport(CSerializedNetMsg& msg, std::vector<unsigned char>& header) = 0;
+    virtual ~TransportSerializer() {}
+};
+
+class V1TransportSerializer  : public TransportSerializer {
+public:
+    void prepareForTransport(CSerializedNetMsg& msg, std::vector<unsigned char>& header);
+};
 
 /** Information about a peer */
 class CNode
 {
     friend class CConnman;
 public:
+    std::unique_ptr<TransportSerializer> m_serializer;
```

Comment:
This change would mean we initialize serializer at `ConnectNode`/`AcceptConnection`, if we make an outbound connection, thanks to `NODE_P2P_V2` we can be sure to have right serializer, but in case of inbound we may need to update it. Maybe comment this logic here?

Reply:
Thanks for the comment.
Your concerns are valid but not part of this PR.
For a BIP324 (NODE_P2P_V2) inbound connection (also outbound) we always start with a V1 serializer/deserializer to perform the handshake. I think the handshake/a.k.a. protocol upgrade belongs to the V1 protocol.

Once the handshake has been completed, we exchange the node objects serializer/deserializer with a V2 instance.

Reply:
You mean the handshake belongs to the V1 protocol because we are using V1 messages to exchange 32-byte public keys between initiator/responder as defined in V2 BIP ? That makes sense and it's already implicit in the BIP, "After a successful handshake, messages MUST use the "v2 messages structure", but maybe put a word about this at the beginning of handshake section, just to be really obvious.